core/src/main/java/org/sonarsource/sonarlint/core/VersionSoonUnsupportedHelper.java

```diff
@@ -0,0 +1,113 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core;
+
+import com.google.common.eventbus.Subscribe;
+import java.util.Set;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.sonarsource.sonarlint.core.clientapi.SonarLintClient;
+import org.sonarsource.sonarlint.core.clientapi.client.message.MessageType;
+import org.sonarsource.sonarlint.core.clientapi.client.message.ShowSoonUnsupportedMessageParams;
+import org.sonarsource.sonarlint.core.commons.ConnectionKind;
+import org.sonarsource.sonarlint.core.commons.Version;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+import org.sonarsource.sonarlint.core.event.BindingConfigChangedEvent;
+import org.sonarsource.sonarlint.core.event.ConfigurationScopesAddedEvent;
+import org.sonarsource.sonarlint.core.repository.config.ConfigurationRepository;
+import org.sonarsource.sonarlint.core.repository.connection.ConnectionConfigurationRepository;
+import org.sonarsource.sonarlint.core.serverapi.system.ServerInfo;
+import org.sonarsource.sonarlint.core.serverconnection.VersionUtils;
+
+@Named
+@Singleton
+public class VersionSoonUnsupportedHelper {
+
+  private static final String UNSUPPORTED_NOTIFICATION_ID = "sonarlint.unsupported.%s.id";
+  private static final String NOTIFICATION_MESSAGE = "The version used by the current connection '%s' will be soon unsupported. " +
+    "Please consider upgrading to the latest %s LTS version to ensure continued support and access to the latest features.";
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+  private final SonarLintClient client;
+  private final ConfigurationRepository configRepository;
+  private final ConnectionConfigurationRepository connectionRepository;
+  private final ServerApiProvider serverApiProvider;
+
+  public VersionSoonUnsupportedHelper(SonarLintClient client, ConfigurationRepository configRepository,
+    ConnectionConfigurationRepository connectionRepository, ServerApiProvider serverApiProvider) {
+    this.client = client;
+    this.configRepository = configRepository;
+    this.connectionRepository = connectionRepository;
+    this.serverApiProvider = serverApiProvider;
+  }
+
+  @Subscribe
+  public void configurationScopesAdded(ConfigurationScopesAddedEvent event) {
+    var configScopeIds = event.getAddedConfigurationScopeIds();
+    notifyIfSoonUnsupported(configScopeIds);
```

Comment:
When adding many config scopes in a row (for example when opening multiple projects in SLE workspace), the client will receive many notifications (one per project), while I would expect to receive only one for the connection.

Reply:
Good point, I will gather the same connections together